core/src/main/java/nu/yona/server/analysis/entities/DayActivity.java

```diff
@@ -70,8 +84,104 @@ public void addActivity(Activity activity)
 		this.activities.add(activity);
 	}
 
+	@Override
+	protected List<Integer> computeSpread()
```

Comment:
Just a quick comment (I need more time to review in detail): this method is too long.